### PR TITLE
popUp window for camera view 

### PR DESCRIPTION
**Pull request by JinaDawood**
Created a mqtt handler class, which handles the mqtt connection and related method when publishing and subscribing in order to avoid having redundant code in each activity when needing mqtt connection.  

Implemented a popup window, which is viewable when pressing the camera button on the manual controlling page. The aim is to display the camera view in that popup window. Which is something we'll keep working on.

### CODE REVIEW

**Comment by JinaDawood** [comment]
> I am not sure if the error is included in this log. Which part of this log is in red color? If the app crashes there should be a stack trace, with an **exception** of some sort that tells you more about what went wrong.
> 
> Also, why are there two places that handle the MQTT connection? Which of the two is actually used?

the app does not really crash, but when I add line 132 in PopUpWindow.kt I cant access that activity anymore. So I get sent back to the start page. 

Yeah right know its a bit messy, Its being handled in both places. Usually the ManualOption is the one that i supposed to handle mqtt, but I added it to PopUpWindow, since I wanted to access the camera

File: .idea/libraries/Gradle__androidx_activity_activity_1_1_0_aar.xml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 <component name="libraryTable">
```
Comment (platisd) [inline .idea/libraries/Gradle__androidx_activity_activity_1_1_0_aar.xml:1]: You shouldn't be committing these files. Look into what files should be included in `gitignore` and remove them if they are already version controlled.

Comment (JinaDawood) [reply .idea/libraries/Gradle__androidx_activity_activity_1_1_0_aar.xml]: yes. Im deleting them.

**Comment by JinaDawood** [comment]
> Not the best person to ask for android/Kotlin stuff, so I may be missing something obvious, but the entire PopUpWindow class seems to be treated in a strange way. Specifically, you create it in MainActivity and you call its showPopUpWindow function and then inside that very function you create another instance of it?
> Why is that?
> My suspicion is that something is going wrong with the overall implementation of that PopUpWindow class but I don't have enough Kotlin knowledge or insight in your codebase to figure out what exactly.

Yeah I understand what you mean. But I think the problem is related to me trying to connect to mqtt. So I kind of did what you said(if i understood it correctly). I created a mqtthandler class which handles the connection and the methods related to publishing/subscribing etc. Then I tried calling it to the manual control class. And the same thing happens. My screen just shuts down. When I remove "mqttHandler!!.connectToMqttBroker(message)" then the screen doesnt shut down, but no connection happens to the mqtt. I've tried 100 several ways with this. I also did it in java, since kotlin is quite hard. But the grade files will not build java since its a kotlin project. So I changed it back

**Comment by platisd** [comment]
If something it doesn't work, you cannot find what is wrong or someone to help you with it, then try a different approach. Skip the pop up window concept and use a normal image view that's included in the same screen as t he rest. Don't let it block you.

**Review by gusandalce — APPROVED** [review]
Seems to be working, will continue to look for bugs as we go along

**Pull request merged**